httpie/resolve.py

```diff
@@ -0,0 +1,53 @@
+import socket
+
+
+class CustomResolver:
+    def __init__(self, resolve):
+        """
+        :param resolve: A list of HOST[:PORT]:ADDRESS[:ADDRESS]? records
+        :type resolve: list[str]
+        """
+        self.entries = {}
+        for entry in resolve:
+            host, port_and_addresses = entry.split(':', 1)
+            if ':' in port_and_addresses:
```

Comment:
For this and the next comment, would something like:
```
    @staticmethod
    def parse_resolve_entry(entry):
        """
        :param entry:
        :return: host, port (or `None` if absent) and
        """
        host, port_and_addresses = entry.split(':', 1)
        try:
            # If port is not absent, e.g. 'example.org:80:127.0.0.1'
            port, addresses = port_and_addresses.split(':', 1)
            port = int(port)
        except ValueError:
            port = None
            addresses = port_and_addresses

        return host, port, [
            a[1:-1] if a.startswith('[') and a.endswith(']') else a
            for a in addresses.split(',')
        ]

    def __init__(self, resolve):
        """
        :param resolve: A list of HOST[:PORT]:ADDRESS[:ADDRESS]? records
        :type resolve: list[str]
        """
        self.entries = {
            (host, port): addresses
            for host, port, addresses in map(self.parse_resolve_entry, resolve)
        }
```
work to make the logic more clear?

Reply:
There's no need/benefit to having this be a static method over a global function. Other than that, I think it makes sense.

Reply:
Separate function is for splitting what's going inside of `__init__` into actual parsing (the function itself) and the internal state of the object. Otherwise, `__init__` is violating the principle of a single responsibility.

Reply:
I was saying there is no need for `parse_resolve_entry` to be a stactic method. It can be a global function. That is what I was saying.

```python
def _parse_resolve_entry(...):
  # ...

class CustomResolver:
```

Static methods are usually used as part of the interface to the class, but it's unexpected for anyone to be calling `CustomResolver.parse_resolve_entry(...)`.